src/Handler/DownloadHandler.php

```diff
@@ -70,7 +70,8 @@ private function createDownloadResponse($stream, string $filename, ?string $mime
 
         $disposition = $response->headers->makeDisposition(
             $forceDownload ? ResponseHeaderBag::DISPOSITION_ATTACHMENT : ResponseHeaderBag::DISPOSITION_INLINE,
-            $filename
+            $filename,
+            'download'.strrchr($filename, '.')
```

Comment:
This looks weak to me.
Can't we use something like `filter_var($filename, FILTER_SANITIZE_STRING, FILTER_FLAG_STRIP_HIGH)`?

Reply:
Most browsers will use UTF-8 version it seems to me so I don't think we really need real file name as fallback. Ofc I can change this code but I don't know how FILTER_FLAG_STRIP_HIGH exactly works, I can't find exact documentation for this flag. From the other hand, here are HeaderUtils requirements: https://github.com/symfony/http-foundation/blob/master/HeaderUtils.php#L174 We can change hardcoded 'download` string to something better if we make sure alternative implementation won't cause another HeaderUtils exception.

Reply:
you can try on the string you provided in the issue. It just strips non ascii away

Reply:
Yes, I tried. But still I am not sure what exactly does FILTER_FLAG_STRIP_HIGH so I can't check whether its output matches HeaderUtils requirements. Keep in mind that when user tries to download file, its much better to download "download.ext" file than to get 500 server error. :) Also, most users will never see fallback filename, most browsers support UTF-8 filenames. 

I will try to look at PHP source code when I will find more time, although.

Reply:
It just removes ascii codes higher than 127

Reply:
@garak  Unfortunately FILTER_FLAG_STRIP_HIGH is not enough because /, \ and % are disallowed by HeaderUtils::makeDisposition(). Slashes are disallowed even in UTF-8 filename. I updated my pull request to filter it manually.